### PR TITLE
Restructuring and cleaning

### PR DESCRIPTION
**Pull request by varma-adithya**
Add Stockwatch.Model and Stockwatch.Winforms. Modified Stockwatch to Stockwatch.Business for service functionalities. Remove umwanted filed from git repo. Add enum for stock symbols.

### CODE REVIEW

File: Stockwatch/Stockwatch.Business/Stockdataservices.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿using Stockwatch.Model;
+
+namespace Stockwatch.Business
+{
+    public class Stockdataservices
```
Comment (AB-GR) [inline Stockwatch/Stockwatch.Business/Stockdataservices.cs:5]: Use pascal naming convention in C# StockDataServices

File: Stockwatch/Stockwatch.Business/Stockdataservices.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        private readonly StockwatchDbContext _context;
+
+        public Stockdataservices(StockwatchDbContext context) { _context = context; }
+
+        public void AddStock(Stockdata stockdata)
```
Comment (AB-GR) [inline Stockwatch/Stockwatch.Business/Stockdataservices.cs:11]: Pascal case StockData

File: Stockwatch/Stockwatch.Model/Stockdata.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿namespace Stockwatch.Model
+{
+    public class Stockdata
```
Comment (AB-GR) [inline Stockwatch/Stockwatch.Model/Stockdata.cs:3]: StockAlertRange would be a better name

File: Stockwatch/Stockwatch.Business/Stockpriceservices.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
+﻿using Newtonsoft.Json.Linq;
+using Stockwatch.Model;
+
+namespace Stockwatch.Business
+{
+    public class Stockpriceservices
```
Comment (AB-GR) [inline Stockwatch/Stockwatch.Business/Stockpriceservices.cs:6]: Pascal name - StockPriceService, also use singular suffix `Service`

File: Stockwatch/Stockwatch.Business/Stockpriceservices.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        public Stockpriceservices(ILogger<Worker> logger)
+        {
+                _logger = logger;
+        }
+
+
+        public async void GetStockprice(Stockdata stockdata)
+        {
+            var apiKey = "EO4BB53HAZMW6TDW";
+            string apiUrl = $"https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol={stockdata.Stocksymbol.SymbolName}&apikey={apiKey}";
```
Comment (AB-GR) [inline Stockwatch/Stockwatch.Business/Stockpriceservices.cs:20]: This should go to app settings

File: Stockwatch/Stockwatch.Business/Stockpriceservices.cs [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+        public Stockpriceservices(ILogger<Worker> logger)
+        {
+                _logger = logger;
+        }
+
+
+        public async void GetStockprice(Stockdata stockdata)
+        {
+            var apiKey = "EO4BB53HAZMW6TDW";
```
Comment (AB-GR) [inline Stockwatch/Stockwatch.Business/Stockpriceservices.cs:19]: This should go to user secrets

File: Stockwatch/Stockwatch.Business/Stockpriceservices.cs [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        private readonly ILogger<Worker> _logger;
+        public IntraStockprice intraStockprice { get; set; }
+
+        public Stockpriceservices(ILogger<Worker> logger)
+        {
+                _logger = logger;
+        }
+
+
+        public async void GetStockprice(Stockdata stockdata)
```
Comment (AB-GR) [inline Stockwatch/Stockwatch.Business/Stockpriceservices.cs:17]: pascal case

Comment (AB-GR) [inline Stockwatch/Stockwatch.Business/Stockpriceservices.cs:17]: should be async Task

Comment (AB-GR) [inline Stockwatch/Stockwatch.Business/Stockpriceservices.cs:17]: Why is named GetStockPrice when this method returns nothing

File: Stockwatch/Stockwatch.Business/Stockpriceservices.cs [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+                        }
+                        else if(stockPrice.Price <= stockdata.Lowerlimit)
+                        {
+                            _logger.LogInformation("Buy more stock!");
+                        }
+                        else { _logger.LogInformation("Keep the stock"); }
+
+                        Console.WriteLine("Success");
+                        Console.WriteLine(stockPrice.Symbol);
+                        Console.WriteLine(stockPrice.Price);
```
Comment (AB-GR) [inline Stockwatch/Stockwatch.Business/Stockpriceservices.cs:48]: Remo Console.WriteLine

File: Stockwatch/Stockwatch.Business/Stocksymbolservices.cs [hunk 9]
```diff
@@ -0,0 +1,28 @@
+﻿using Stockwatch.Model;
+
+namespace Stockwatch.Business
+{
+    public class Stocksymbolservices
```
Comment (AB-GR) [inline Stockwatch/Stockwatch.Business/Stocksymbolservices.cs:5]: Pascal case

File: Stockwatch/Stockwatch.Business/Worker.cs [hunk 10]
```diff
@@ -1,6 +1,6 @@
-using Microsoft.Extensions.Hosting;
+using Stockwatch.Model;
 
-namespace Stockwatch
+namespace Stockwatch.Business
 {
     public class Worker : BackgroundService
```
Comment (AB-GR) [inline Stockwatch/Stockwatch.Business/Worker.cs:5]: Separate out BackgroundService from Business project

File: Stockwatch/Stockwatch.Business/appsettings.Development.json [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+      "Microsoft": "Warning",
+      "Microsoft.Hosting.Lifetime": "Information"
+    }
+  },
+  "EventLog": {
+    "LogLevel": {
+      "Default": "Information",
+      "Microsoft.Hosting.Lifetime": "Information"
+    }
+  }
```
Comment (AB-GR) [inline Stockwatch/Stockwatch.Business/appsettings.Development.json:14]: EventLog resides outside Logging

File: Stockwatch/Stockwatch.Business/Stocksymbolservices.cs (hunk 9 above)
Comment (AB-GR) [inline Stockwatch/Stockwatch.Business/Stocksymbolservices.cs:5]: Service suffix should be singular

File: Stockwatch/Stockwatch.Business/Stockwatch.Business.csproj [hunk 12]
```diff
@@ -8,7 +8,14 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="7.0.11" />
     <PackageReference Include="Microsoft.Extensions.Hosting" Version="6.0.1" />
     <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="7.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Logging.EventLog" Version="7.0.0" />
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
```
Comment (AB-GR) [inline Stockwatch/Stockwatch.Business/Stockwatch.Business.csproj:15]: System.Text.Json should be used

File: Stockwatch/Stockwatch.Business/Worker.cs [hunk 13]
```diff
@@ -27,7 +28,11 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
             while (!stoppingToken.IsCancellationRequested)
             {
                 _logger.LogInformation("Worker running at: {time}", DateTimeOffset.Now);
-                await Task.Delay(1000, stoppingToken);
+                var stksymb = new Stocksymbol { SymbolName = "Appl" }; ;
+                var stk = new Stockdata { Stocksymbol = stksymb, SymbolId = stksymb.Id , Upperlimit = 200, Lowerlimit = 50};
+                var stkdata = new Stockpriceservices(_logger);
+                stkdata.GetStockprice(stk);
```
Comment (AB-GR) [inline Stockwatch/Stockwatch.Business/Worker.cs:34]: All of these should come via DI

File: Stockwatch/Stockwatch.Model/DbContextFactory.cs [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+using Microsoft.EntityFrameworkCore.Design;
+
+namespace Stockwatch.Model
+{
+    public class StockwatchDbContextFactory : IDesignTimeDbContextFactory<StockwatchDbContext>
+    {
+        public StockwatchDbContext CreateDbContext(string[] args)
+        {
+            var builder = new DbContextOptionsBuilder<StockwatchDbContext>();
+            var connectionString = "Data Source=..\\stock_database.db";
```
Comment (AB-GR) [inline Stockwatch/Stockwatch.Model/DbContextFactory.cs:11]: Before deploying decide where database should be, a location common to both worker & ui

File: Stockwatch/Stockwatch.Model/Stockdata.cs [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+    {
+        public int Id { get; set; }
+        public int SymbolId { get; set; }
+        public decimal Upperlimit { get; set; }
```
Comment (AB-GR) [inline Stockwatch/Stockwatch.Model/Stockdata.cs:7]: UpperLimit, LowerLimit

File: Stockwatch/Stockwatch.Model/Stocksymbol.cs [hunk 16]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Stockwatch.Model
+{
+    public class Stocksymbol
```
Comment (AB-GR) [inline Stockwatch/Stockwatch.Model/Stocksymbol.cs:3]: Pascal case

File: Stockwatch/Stockwatch.Business/Stockdataservices.cs (hunk 1 above)
Comment (varma-adithya) [reply Stockwatch/Stockwatch.Business/Stockdataservices.cs:5]: Done

File: Stockwatch/Stockwatch.Business/Stockdataservices.cs (hunk 2 above)
Comment (varma-adithya) [reply Stockwatch/Stockwatch.Business/Stockdataservices.cs:11]: Done

File: Stockwatch/Stockwatch.Business/Stockpriceservices.cs (hunk 4 above)
Comment (varma-adithya) [reply Stockwatch/Stockwatch.Business/Stockpriceservices.cs:6]: Done

File: Stockwatch/Stockwatch.Business/Stockpriceservices.cs (hunk 7 above)
Comment (varma-adithya) [reply Stockwatch/Stockwatch.Business/Stockpriceservices.cs:17]: Done

File: Stockwatch/Stockwatch.Business/Stockpriceservices.cs (hunk 8 above)
Comment (varma-adithya) [reply Stockwatch/Stockwatch.Business/Stockpriceservices.cs:48]: Done

File: Stockwatch/Stockwatch.Business/Stocksymbolservices.cs (hunk 9 above)
Comment (varma-adithya) [reply Stockwatch/Stockwatch.Business/Stocksymbolservices.cs:5]: Done

File: Stockwatch/Stockwatch.Business/appsettings.Development.json (hunk 11 above)
Comment (varma-adithya) [reply Stockwatch/Stockwatch.Business/appsettings.Development.json:14]: Done

File: Stockwatch/Stockwatch.Business/Stocksymbolservices.cs (hunk 9 above)
Comment (varma-adithya) [reply Stockwatch/Stockwatch.Business/Stocksymbolservices.cs:5]: Done

File: Stockwatch/Stockwatch.Model/Stocksymbol.cs (hunk 16 above)
Comment (varma-adithya) [reply Stockwatch/Stockwatch.Model/Stocksymbol.cs:3]: Done

File: Stockwatch/Stockwatch.Model/Stockdata.cs (hunk 15 above)
Comment (varma-adithya) [reply Stockwatch/Stockwatch.Model/Stockdata.cs:7]: Done

File: Stockwatch/Stockwatch.Business/Stockwatch.Business.csproj (hunk 12 above)
Comment (varma-adithya) [reply Stockwatch/Stockwatch.Business/Stockwatch.Business.csproj:15]: Used

File: Stockwatch/Stockwatch.Business/Stockpriceservices.cs (hunk 7 above)
Comment (varma-adithya) [reply Stockwatch/Stockwatch.Business/Stockpriceservices.cs:17]: Done

Comment (varma-adithya) [reply Stockwatch/Stockwatch.Business/Stockpriceservices.cs:17]: Return IntraStockPrice which is the updated details of the requested stock

**Pull request merged**